### PR TITLE
website: Redux store enhancements & organization

### PR DESCRIPTION
**Pull request by OchiengPaul442**
#### Summary of Changes (What does this PR do?)

- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

#### Status of maturity (all need to be checked before merging):

- [ ] I've tested this locally
- [ ] I consider this code done
- [ ] This change ready to hit production in its current state

#### Screenshots (optional)
image
image

## Summary by CodeRabbit

- **New Features**
	- Enhanced environment variable management by activating loading from `.env` files.
	- Improved organization of installed applications for better readability.
	- Optimized Docker build process with a multi-stage approach for frontend and backend.
	- Integrated state persistence in Redux for improved user experience.
	- Added performance optimization plugins in the webpack configuration for better build efficiency.
	- Updated localization keys for clarity and added new placeholders for user input fields.

- **Bug Fixes**
	- Improved error handling in asynchronous data fetching across various slices.

- **Refactor**
	- Simplified Redux store configuration and modularized slice imports for better maintainability.
	- Enhanced modal state management for improved flexibility and future extensibility.
	- Improved styling consistency and readability in various SCSS files.
	- Refactored component state access to align with standard Redux practices.

### CODE REVIEW

File: website/frontend/src/components/CleanAir/Hightlights/MainHighlight.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import React, { useEffect } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
-import { getAllEvents } from '../../../../reduxStore/Events/EventSlice';
+import { getAllEvents } from '../../../../reduxStore/Events';
```
Comment (Baalmart) [inline website/frontend/src/components/CleanAir/Hightlights/MainHighlight.js:3]: Hi @OchiengPaul442 , I hope we can have simple import statements, these lenghty backward paths can be prone to human errors. :) I remember we had some sort of library to support simpler paths?

File: website/requirements.txt [hunk 2]
```diff
@@ -39,6 +39,7 @@ pytz==2024.1
 PyYAML==6.0.1
 requests==2.31.0
 rsa==4.9
+setuptools
```
Comment (Baalmart) [inline website/requirements.txt:42]: this does not have a version number, intentional? @OchiengPaul442

**Review by Baalmart — COMMENTED** [review]
Thanks @OchiengPaul442 , I have left a few comments for your review.

File: website/frontend/src/components/CleanAir/Hightlights/MainHighlight.js (hunk 1 above)
Comment (OchiengPaul442) [reply website/frontend/src/components/CleanAir/Hightlights/MainHighlight.js:3]: thanks @Baalmart,
we do have the library installed, though I might have forgotten and used the relative paths in this case.

File: website/requirements.txt (hunk 2 above)
Comment (OchiengPaul442) [reply website/requirements.txt:42]: If no version number is specified, the latest stable version of the package will be used. This ensures that the most up-to-date, stable version of the package is installed.

**Review by Baalmart — APPROVED** [review]
thanks @OchiengPaul442

**Pull request merged**